core/src/main/java/org/apache/iceberg/io/OutputFileFactory.java

```diff
@@ -95,25 +100,71 @@ public OutputFileFactory(PartitionSpec spec, FileFormat format, LocationProvider
     this.operationId = operationId;
   }
 
+  public static Builder builderFor(Table table, FileFormat format, int partitionId, long taskId) {
```

Comment:
It is a bit debatable what we consider as required arguments. I think these 4 are always set.

We may consider defaulting the format to the current table format for data files but we have to ensure it is the same format we use in data writers. That's why it safer to always require an explicit value.

We can default partition and task IDs to 1 just for tests. These values must be always set in real scenarios.

Reply:
What about removing format and defaulting that using the table property? Then if an engine wants to override it can, but otherwise it is automatic.

Reply:
+1 for having default file format from the table properties, so that when `write-format` option is not set it can default to the table default instead of finding it again and pass it to the call.

And btw it's probably a good idea to have a util method to get the file format from the properties map, since we are doing it everywhere, but no need to have that in this PR.

Reply:
Hopefully this will standardize that util method so that only this class is needed.

Reply:
I moved the format from the list of required args. Let's consider adding a utility method once we have clarity on how the writer factory will look like and how it will be used.